File: src/main/java/com/paragon/client/ui/console/ConsoleGUI.java
package com.paragon.client.ui.console;

import com.paragon.Paragon;
import com.paragon.client.systems.module.impl.client.ClickGUI;
import net.minecraft.client.gui.GuiScreen;

import java.io.IOException;

public class ConsoleGUI extends GuiScreen {

    @Override
    public void initGui() {
        Paragon.INSTANCE.getConsole().init();
        super.initGui();
    }

    @Override
    public void drawScreen(int mouseX, int mouseY, float partialTicks) {
        // Make the background darker
        if (ClickGUI.darkenBackground.getValue()) {
            drawDefaultBackground();
        }

        Paragon.INSTANCE.getConsole().draw(mouseX, mouseY);

        Paragon.INSTANCE.getTaskbar().drawTaskbar(mouseX, mouseY);

        super.drawScreen(mouseX, mouseY, partialTicks);
    }

    @Override
    protected void mouseClicked(int mouseX, int mouseY, int mouseButton) throws IOException {
        Paragon.INSTANCE.getTaskbar().mouseClicked(mouseX, mouseY);

        Paragon.INSTANCE.getConsole().mouseClicked(mouseX, mouseY, mouseButton);

        super.mouseClicked(mouseX, mouseY, mouseButton);
    }

    @Override
    protected void keyTyped(char typedChar, int keyCode) throws IOException {
        Paragon.INSTANCE.getConsole().keyTyped(typedChar, keyCode);

        super.keyTyped(typedChar, keyCode);
    }

    @Override
    public boolean doesGuiPauseGame() {
        return ClickGUI.pause.getValue();
    }
}

File: src/main/kotlin/com/paragon/impl/ui/console/ConsoleGUI.kt
package com.paragon.impl.ui.console

import com.paragon.Paragon
import com.paragon.impl.module.client.ClickGUI
import com.paragon.impl.ui.util.Click.Companion.getClick
import net.minecraft.client.gui.GuiScreen
import java.io.IOException

/**
 * @author Surge
 */
class ConsoleGUI : GuiScreen() {

    override fun initGui() {
        Paragon.INSTANCE.console.init()
        super.initGui()
    }

    override fun drawScreen(mouseX: Int, mouseY: Int, partialTicks: Float) {
        // Make the background darker
        if (ClickGUI.darkenBackground.value) {
            drawDefaultBackground()
        }

        Paragon.INSTANCE.console.draw(mouseX, mouseY)
        Paragon.INSTANCE.taskbar.draw(mouseX, mouseY)

        super.drawScreen(mouseX, mouseY, partialTicks)
    }

    @Throws(IOException::class)
    override fun mouseClicked(mouseX: Int, mouseY: Int, mouseButton: Int) {
        Paragon.INSTANCE.taskbar.mouseClicked(mouseX, mouseY, getClick(mouseButton))
        Paragon.INSTANCE.console.mouseClicked(mouseX, mouseY, mouseButton)

        super.mouseClicked(mouseX, mouseY, mouseButton)
    }

    @Throws(IOException::class)
    override fun keyTyped(typedChar: Char, keyCode: Int) {
        Paragon.INSTANCE.console.keyTyped(typedChar, keyCode)

        super.keyTyped(typedChar, keyCode)
    }

    override fun doesGuiPauseGame() = ClickGUI.pause.value

}